examples/Atlas/+atlasControllers/@InstantaneousQPController/InstantaneousQPController.m

```diff
@@ -0,0 +1,328 @@
+% for (time-varing) linear COM/ZMP dynamics. Includes logic specific to
+% atlas/bipeds for raising the heel while walking. This differs from
+% AtlasQPController in that it contains no stateful information about the
+% currently executed plan. Instead, it is designed to be fully general for a
+% variety of plan types (standing, walking, manipulating, etc.). The
+% AtlasPlanEval class now handles the state of the current plan.
+  properties(SetAccess=private, GetAccess=public);
+    debug;
+    debug_pub;
+    robot;
+    controller_data
+    q_integrator_data
+    vref_integrator_data
+    robot_property_cache
+    use_mex;
+    mex_ptr;
+    support_detect_mex_ptr;
+    use_bullet = false;
+    default_terrain_height = 0;
+    param_sets
+    gurobi_options = struct();
+    solver = 0;
+  end
+
+  properties
+    quiet = true;
+  end
+
+  methods
+    function obj = InstantaneousQPController(r, param_sets, options)
+      if nargin < 3
+        options = struct();
+      end
+      if nargin < 2 || isempty(param_sets)
+        param_sets = atlasParams.getDefaults(r);
+      end
+      options = applyDefaults(options,...
+        struct('debug', false,...
+               'use_mex', 1, ...
+               'solver', 0),...
+        struct('debug', @(x) typecheck(x, 'logical') && sizecheck(x, 1),...
+               'use_mex', @isscalar, ...
+               'solver', @(x) x == 0 || x == 1));
+      for f = fieldnames(options)'
+        obj.(f{1}) = options.(f{1});
+      end
+
+      obj.robot = r;
+      obj.param_sets = param_sets;
+      obj.robot_property_cache = atlasUtil.propertyCache(r);
+      import atlasControllers.*;
+      import atlasFrames.*;
+
+      if obj.debug
+        obj.debug_pub = ControllerDebugPublisher('CONTROLLER_DEBUG');
+      end
+
+      obj.controller_data = InstantaneousQPControllerData(struct('infocount', 0,...
+                                                     'qp_active_set', []));
+      obj.q_integrator_data = IntegratorData(r);
+      obj.vref_integrator_data = VRefIntegratorData(r);
+
+      obj.gurobi_options.outputflag = 0; % not verbose
+      if obj.solver==0
+        obj.gurobi_options.method = 2; % -1=automatic, 0=primal simplex, 1=dual simplex, 2=barrier
+      else
+        obj.gurobi_options.method = 0; % -1=automatic, 0=primal simplex, 1=dual simplex, 2=barrier
+      end
+      obj.gurobi_options.presolve = 0;
+      % obj.gurobi_options.prepasses = 1;
+
+      if obj.gurobi_options.method == 2
+        obj.gurobi_options.bariterlimit = 20; % iteration limit
+        obj.gurobi_options.barhomogeneous = 0; % 0 off, 1 on
+        obj.gurobi_options.barconvtol = 5e-4;
+      end
+
+      terrain = getTerrain(r);
+      obj.default_terrain_height = r.getTerrainHeight([0;0]);
+      if isa(terrain,'DRCTerrainMap')
+        terrain_map_ptr = terrain.map_handle.getPointerForMex();
+      else
+        terrain_map_ptr = 0;
+      end
+
+      obj.mex_ptr = ...
```

Comment:
perhaps rename this to data_mex_ptr?  (the mex name and the pointer name feel like they imply two different things)